sdk/go/x/auto/local_workspace.go

```diff
@@ -584,6 +584,14 @@ type GitRepo struct {
 	CommitHash string
 	// Optional function to execute after enlisting in the specified repo.
 	Setup SetupFn
+	// The absolute path to a private key for access to the github repo
+	SSHPrivateKeyPath string
+	// The password that pairs with a username or as part of an SSH Private Key
+	Password string
+	// PersonalAccessToken is a GitHub personal access token is in replacment of your password
```

Comment:
nit: is there any reason to use github over the more generic `git` (here and comments above)? I believe gitlab and others support similar auth mechanisms.